### ISSUE TITLE
Restore Failed on Domains with Mail Setups using SSL

### ISSUE THREAD

**Issue by brackenhill-mob** [hestiacp/hestiacp#1069]
Hestia v1.2.3 on both old and new servers. Old server running Ubuntu 18.04 LTS and new Ubuntu 20.04 LTS

I am migrating our Hestia deployment to a bigger server. The admin a/c contains the domains with mail setups. Everything restores correctly on all domains except those with a mail setup that uses SSL (Let’s Encrypt) - mail domains without SSL are OK.

The issue is that none of the SSL data is restored. This includes the Apache and nginx conf files that should be in /home/admin/conf/mail/DOMAIN and the SSL certificates themselves - both in the admin account and the mail.DOMAIN.* SSL files in the /hestia/ssl folder in the tar file. The files exist in the tar backup.

In the Hestia error log there are multiple Error 3’s for the restore command. For each affected domain I received these errors on screen

<pre>
chmod: cannot access '/usr/local/hestia/data/users/admin/ssl/mail.XXX.com.*': No such file or directory
cp: cannot stat '/usr/local/hestia/data/users/admin/ssl/mail.XXX.com.crt': No such file or directory
cp: cannot stat '/usr/local/hestia/data/users/admin/ssl/mail.XXX.com.key': No such file or directory
cp: cannot stat '/usr/local/hestia/data/users/admin/ssl/mail.XXX.com.pem': No such file or directory
/usr/local/hestia/bin/v-restore-user: line 645: /etc/dovecot/conf.d/domains/XXX.com.conf: No such file or directory
/usr/local/hestia/bin/v-restore-user: line 646: /etc/dovecot/conf.d/domains/XXX.com.conf: No such file or directory
/usr/local/hestia/bin/v-restore-user: line 647: /etc/dovecot/conf.d/domains/XXX.com.conf: No such file or directory
/usr/local/hestia/bin/v-restore-user: line 648: /etc/dovecot/conf.d/domains/XXX.com.conf: No such file or directory
/usr/local/hestia/bin/v-restore-user: line 649: /etc/dovecot/conf.d/domains/XXX.com.conf: No such file or directory
find: '/usr/local/hestia/ssl/mail': No such file or directory
ln: failed to create symbolic link '/usr/local/hestia/ssl/mail/mail.XXX.com.crt': No such file or directory
ln: failed to create symbolic link '/usr/local/hestia/ssl/mail/mail.XXX.com.key': No such file or directory
chmod: cannot access '/home/admin/conf/mail/XXX.com/ssl/*': No such file or directory
chown: cannot access '/home/admin/conf/mail/XXX.com/ssl/*': No such file or directory
chmod: cannot access '/usr/local/hestia/ssl/mail/*': No such file or directory
chown: cannot access '/usr/local/hestia/ssl/mail/*': No such file or directory
</pre>

The notification message in HestiaCP (on the bell icon) says that the restore was successful when clearly it is not!

Given the messages about missing files I tried re-running v-restore-user with the optional MAIL switch. This time there were no on screen errors but again the SSL files failed to be restored.

**Comment by FalzoMAD** [comment]
can confirm, that on an empty/fresh install (used ubuntu 18.04) the restore fails. there seems to be two different issue here though.

`/usr/local/hestia/bin/v-restore-user: line 645: /etc/dovecot/conf.d/domains/XXX.com.conf: No such file or directory`

for the errors about dovcot the /etc/dovecot/conf.d/**domains** directory seems to be missing on a fresh install, most likely this is only created when adding the very first mail-domain in the panel but not during install or if a restore is the very first action.

if the folder is created/available, the restore to a normal (non-existing) user works without issues. using the admin user to host domains/mails is not recommended anyway. however, with the dovecot part resolved the issue about the missing files in the admin data folder still persists. 

therefore I tried to create a user first and with that the same issue appears also for that existing (empty) user. from a glimpse at the restore script I guess there is an issue with writing to the mail.conf file (if it exists) and there it can't grab the SSL=YES later on...

TL;DR; two issues, existence of folder /etc/dovecot/conf.d/domains has to be checked (and evtl. created). writing data to mail.conf for existing users seems to fail.

**Comment by jaapmarcus** [comment]
Just tested and can confirm the issue. 

This should fix the issue if somebody can test it would be appreciated 

https://github.com/hestiacp/hestiacp/commit/25e448adfcb2dca741da6829f0b4060a0f3cb51e

**Comment by jaapmarcus** [comment]
And merged into staging/fixes will be in next release..
